### PR TITLE
feat: try to load more geofences when scrolling to the bottom of the …

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Added pagination to geofence list
- Added unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added]
- [x] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

File: src/AmplifyGeofenceControl/ui.ts [hunk 1]
```diff
@@ -271,6 +271,12 @@ export function AmplifyGeofenceControlUI(
       "amplify-ctrl-list",
       geofenceListContainer
     );
+    _geofenceList.addEventListener("scroll", () => {
+      const { scrollHeight, scrollTop, clientHeight } = _geofenceList;
+      if (scrollTop + clientHeight >= scrollHeight - 20) {
```
Comment (mtliendo) [inline src/AmplifyGeofenceControl/ui.ts:276]: curious of what happens in an `else` case here. Main comment: Can you add a gif to show the loading of more data when a scroll occurs.

**Review by mtliendo — CHANGES_REQUESTED** [review]
Can you add a gif to show what the UI looks like. It's not clear if there is a spinner that shows up or what kind of feedback the user gets that more data is being fetched.

**Comment by thaddmt** [comment]
> Can you add a gif to show what the UI looks like. It's not clear if there is a spinner that shows up or what kind of feedback the user gets that more data is being fetched.

This was just something quick I put together as I don't know what the UX will be for more data being fetched so there's no loading spinner just yet

**Comment by mtliendo** [comment]
Oh gotcha, makes sense. Just confirming: Is there a card to fast follow on this before launch?

**Comment by thaddmt** [comment]
> Oh gotcha, makes sense. Just confirming: Is there a card to fast follow on this before launch?

Not yet but will create one!

File: src/AmplifyGeofenceControl/ui.ts (hunk 1 above)
Comment (thaddmt) [reply src/AmplifyGeofenceControl/ui.ts:276]: Here if we're just not at the bottom of the page list then we just do nothing, or that's the intention anyways :)

**Comment by thaddmt** [comment]
pagination

File: src/AmplifyGeofenceControl/ui.ts (hunk 1 above)
Comment (thaddmt) [reply src/AmplifyGeofenceControl/ui.ts:276]: Added gif below

**Review by mtliendo — APPROVED** [review]

**Review by TreTuna — APPROVED** [review]

File: src/AmplifyGeofenceControl/ui.ts (hunk 1 above)
Comment (hvergara) [inline src/AmplifyGeofenceControl/ui.ts:276]: why those `20` pixels? can we put it into a constant?

**Pull request merged**